src/main/java/fi/metatavu/metamind/story/StoryController.java

```diff
@@ -362,5 +386,152 @@ public void deleteIntent(Intent intent) {
+    exportedStory.setIntents(intentsToExport.stream().map(intent -> exportIntent(intent)).collect(Collectors.toList()));
+    exportedStory.setTrainingMaterials(trainingMaterialsToExport.stream().map(trainingMaterial -> trainingMaterialController.exportTrainingMaterial(trainingMaterial)).collect(Collectors.toList()));
+    exportedStory.setVariables(variablesToExport.stream().map((variable) -> exportVariable(variable)).collect(Collectors.toList()));
+    
+    return exportedStory;
+  }
+  
+  /**
+   * Exports a knot
+   * 
+   * @param knot to be exported
+   * @return exported knot
+   */
+  public ExportedStoryKnot exportKnot(Knot knot) {
+    ExportedStoryKnot exportedKnot = new ExportedStoryKnot();
+    Coordinates coordinates = new Coordinates();
+    coordinates.setX(knot.getCoordinateX());
+    coordinates.setY(knot.getCoordinateY());
+    exportedKnot.setContent(knot.getContent());
+    exportedKnot.setCoordinates(coordinates);
+    exportedKnot.setId(knot.getId());
+    exportedKnot.setType(knot.getType());
+    exportedKnot.setTokenizer(knot.getTokenizerType());
+    exportedKnot.setName(knot.getName());
+    exportedKnot.setHint(knot.getHint());
+    return exportedKnot;
+  }
+  
+  /**
+   * Exports an intent
+   * 
+   * @param intent to be exported
+   * @return exported intent
+   */
+  public ExportedStoryIntent exportIntent(Intent intent) {
+    ExportedStoryIntent exportedIntent = new ExportedStoryIntent();
+    
+    exportedIntent.setId(intent.getId());
+    exportedIntent.setName(intent.getName());
+    exportedIntent.setQuickResponse(intent.getQuickResponse());
+    exportedIntent.setQuickResponseOrder(intent.getQuickResponseOrder());
+    exportedIntent.setType(intent.getType());
+    exportedIntent.setSourceKnotId(intent.getSourceKnot().getId());
+    exportedIntent.setTargetKnotId(intent.getTargetKnot().getId());
+    exportedIntent.setGlobal(intent.getGlobal());
+    
+    List<TrainingMaterial> intentMaterial = trainingMaterialController.listTrainingMaterialByIntent(intent);
+    exportedIntent.setTrainingMaterialIds(intentMaterial.stream().map(trainingMaterial -> {
+      return trainingMaterial.getId();
+    }).collect(Collectors.toList()));
+    
+    return exportedIntent;
+  }
+  
+  /**
+   * Exports a variable
+   * 
+   * @param variable to be exported
+   * @return exported variable
+   */
+  public ExportedStoryVariable exportVariable(Variable variable) {
+    ExportedStoryVariable exportedVariable = new ExportedStoryVariable();
+    exportedVariable.setName(variable.getName());
+    exportedVariable.setType(variable.getType());
+    exportedVariable.setValidationScript(variable.getValidationScript());
+    return exportedVariable;
+  }
+  
+  /**
+   * Imports a story
+   * 
+   * @param story that was previously exported
+   * @param id of user doing the import
+   * @return imported story
+   */
+  public Story importStory(ExportedStory body, UUID userId) {
```

Comment:
private

Reply:
This method is used elsewhere, hence it should not be private